presto-spi/src/main/java/com/facebook/presto/spi/ClientRequestFilterContext.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi;
+
+import java.security.Principal;
+
+public class ClientRequestFilterContext
+{
+    private Principal principal;
+
+    public ClientRequestFilterContext(Principal principal)
+    {
+        this.principal = principal;
+    }
+
+    public Principal getPrincipal()
+    {
+        return principal;
+    }
+
+    public void setPrincipal(Principal principal)
+    {
+        this.principal = principal;
+    }
+}
```

Comment:
```suggestion
public class ClientRequestFilterContext
{
    private final Principal principal;

    public ClientRequestFilterContext(Principal principal)
    {
        this.principal = principal;
    }

    public Principal getPrincipal()
    {
        return principal;
    }
}
```

1. The setter is not used.
2. As described in our developer guide, make it immutable.